src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -114,6 +126,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return returnValue
         End Function
 
+        Private Shared Function IsFloatPointExpressionOfUnknownPrecision(rewrittenNode As BoundExpression) As Boolean
+            If rewrittenNode Is Nothing Then
+                Return False
+            End If
+
+            Dim Type = rewrittenNode.Type
+            If Type.SpecialType <> SpecialType.System_Double AndAlso Type.SpecialType <> SpecialType.System_Single Then
+                Return False
+            End If
+
+            Select Case rewrittenNode.Kind
+                ' ECMA-335   I.12.1.3 Handling of floating-point data types.
+                '    ... the value might be retained in the internal representation
+                '   for future use, if it is reloaded from the storage location without having been modified ...
+                '
+                ' Unfortunately, the above means that precision is not guaranteed even when loading from storage.
+                '
+                ' Case BoundKind.FieldAccess
+                ' Case BoundKind.ArrayAccess
+                '    Return True
+
+                Case BoundKind.Sequence
```

Comment:
:memo: I didn't not manage to hit this branch, or its sibling in C#. I kept it to match C#.